### ISSUE TITLE
Breaking tag changes & deprecation warning

### ISSUE THREAD

**Issue by mal** [chef/knife-ec2#526]
# Version:

0.18.0 via ChefDK

# Environment:
Chef DK 3.0.36
Ubuntu 18.04

# Scenario:

Was using `--tags` flag with `--tag-node-in-chef` in 0.17.0 and earlier. Starting seeing deprecation warnings in 0.18.0. Expected to have until at least 0.19.0 to make changes to work flow, but it was actually an immediate breaking change as a result of [this code](https://github.com/chef/knife-ec2/pull/520/files#diff-e2993b8367db674f58a052036f00052aR1040).

# Steps to Reproduce:

Use `--tags` flag with `--tag-node-in-chef`, see deprecation warning which suggests while the flags are on the way out, but with an expectation that they still function:
```
WARN: [DEPRECATED] --tags option is deprecated. Use --aws-tag option instead.
WARN: [DEPRECATED] --tag-node-in-chef option is deprecated. Use --chef-tag option instead.
```

# Expected Result:

Flags behave as intended in 0.17.0 and earlier and will cease to exist in a future version.

# Actual Result:

Values are used for AWS tags, but no corresponding Chef tags are set, immediately breaking compat.

# Comment

This isn't a request to revisit this as that doesn't seem very productive, however it would be nice to know if this is something to expect going forward as it differs vastly in behaviour to `chef/chef` where deprecation warnings are warnings that changes are coming, not that the use case has already been broken. Consistency in what to expect when seeing a deprecation between the two would be nice to have.

**Comment by sirosen** [comment]
I just got hit by this.

It would have been much better for `knife ec2 server create` to hard fail, rather than create a bunch of improperly tagged instances which we then had to clean up.
That change can still be made and would be an improvement over the current behavior -- knife-ec2 >=0.18.1 could fail when given `--tags` instead of creating servers but not properly tagging them.

It's a month after this was released, so I don't really expect any action, but I wanted to offer more explanation than a 👍
